src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -612,11 +596,25 @@ private AuthenticationResult processTokenResponse(HttpWebResponse webResponse) {
                 }
                 break;
             default:
-                result = new AuthenticationResult(String.valueOf(statusCode), new String(webResponse.getBody()), null);
+                throw new HttpResponseException(String.valueOf(statusCode), new String(webResponse.getBody()));
```

Comment:
Line 595 above should also be treated as an exception.

https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/588/files/eddc1d8b2f879e4efa48a2cba59c91ad347f789c..96417f1d1e98c4b8aa279f704aadb89086edfe71#diff-e52b28383a3d6563a748d17a7629dd8fR595